src/owl/dense/owl_dense_matrix_generic.ml

```diff
@@ -565,14 +565,14 @@ let of_array k x m n =
   Owl_dense_ndarray_generic.reshape y [|m; n|]
 
 
-let save_txt ?(sep="\t") ?(append=false) x f =
+let save_txt ?(sep="\t") ?(append=false) x ~out =
```

Comment:
Agree with putting matrix as the last element. Just a matter of coding style preference, but perhaps also move the labelled arguments before `x`?  Aside from these parts, this PR looks quite good to me.

Reply:
Similar comment from me, would it cause a lot of troubles if we simply move matrix as the last input argument? I'm fine with keeping out as named argument anyway.

Reply:
I am wary of breaking changes unless they really improve something. In this case I don’t really see the benefit, matrix and output are clearly specified by the type and flipping the argument is  fast with an inline fun declaration when in a pipe

Reply:
yea the reason I kept the labelled argument at the very end is to avoid breaking changes, but it seems like `dune` enforces labels when there are labelled arguments, so perhaps the point is moot anyway.

in terms of benefit, inline function declaration definitely does the job. but it seems like we end up writing so many of these that it might justify the breaking change in my opinion...